public/controllers/register-agent/containers/steps/steps.tsx

```diff
@@ -1,143 +1,104 @@
+import { useRegisterAgentCommands } from '../../hooks/use-register-agent-commands';
+import {
+  osCommandsDefinitions,
+  optionalParamsDefinitions,
+  tOperatingSystem,
+  tOptionalParameters,
+} from '../../core/config/os-commands-definitions';
+import { UseFormReturn } from '../../../../components/common/form/types';
+import CommandOutput from '../../components/command-output/command-output';
+import ServerAddressTitle from '../../components/server-address/server-address-title';
+import ServerAddressInput from '../../components/server-address/server-address-input';
+import OptionalsInputs from '../../components/optionals-inputs/optionals-inputs';
 
-const warningForAgentName =
-  'The agent name must be unique. It can’t be changed once the agent has been enrolled.';
+interface IStepsProps {
+  needsPassword: boolean;
+  hideTextPassword: boolean;
+  form: UseFormReturn;
+  osCard: React.ReactElement;
+  connection: {
+    isUDP: boolean;
+    isSecure: boolean;
+  };
+  wazuhPassword: string;
+}
 
 export const Steps = ({
   needsPassword,
   hideTextPassword,
-  agentGroup,
   form,
   osCard,
-}) => {
-  const [isPopoverServerAddress, setIsPopoverServerAddress] = useState(false);
-  const [isPopoverAgentName, setIsPopoverAgentName] = useState(false);
+  connection,
+  wazuhPassword,
+}: IStepsProps) => {
+  const [registerAgentFormValues, setRegisterAgentFormValues] =
+    useState<IParseRegisterFormValues>({
+      operatingSystem: {
+        name: '',
+        architecture: '',
+      },
+      optionalParams: {
+        agentGroups: '',
+        agentName: '',
+        serverAddress: '',
+        wazuhPassword,
+      },
+    });
 
-  const onButtonServerAddress = () =>
-    setIsPopoverServerAddress(
-      isPopoverServerAddress => !isPopoverServerAddress,
+  useEffect(() => {
+    // get form values and parse them divided in OS and optional params
+    const registerAgentFormValuesParsed = parseRegisterAgentFormValues(
+      getRegisterAgentFormValues(form),
+      OPERATING_SYSTEMS_OPTIONS,
     );
-  const closeServerAddress = () => setIsPopoverServerAddress(false);
+    setRegisterAgentFormValues(registerAgentFormValuesParsed);
+  }, [form.fields]);
+
+  const { installCommand, startCommand, selectOS, setOptionalParams } =
+    useRegisterAgentCommands<tOperatingSystem, tOptionalParameters>({
+      osDefinitions: osCommandsDefinitions,
+      optionalParamsDefinitions: optionalParamsDefinitions,
+    });
+
+  useEffect(() => {
+    if (
+      registerAgentFormValues.operatingSystem.name !== '' &&
+      registerAgentFormValues.operatingSystem.architecture !== ''
+    ) {
+      selectOS(registerAgentFormValues.operatingSystem as tOperatingSystem);
+    }
 
-  const onButtonAgentName = () =>
-    setIsPopoverAgentName(isPopoverAgentName => !isPopoverAgentName);
-  const closeAgentName = () => setIsPopoverAgentName(false);
+    setOptionalParams(registerAgentFormValues.optionalParams);
+  }, [registerAgentFormValues]);
```

Comment:
**thought**: if we use the form state to generate the related install and start command, instead of saving in a new state, we will avoid a rerender.